Logstash/logstash_filebeats.conf

```diff
@@ -1,73 +1,137 @@
 	ruby{
-	     code =>"
+		code =>"
+
+		# method to split the supplied string by comma, trim whitespace and return an array
+		def mapStringToArray(strFieldValue)
 
-		 # method to split the supplied string by comma, trim whitespace and return an array
-		 def mapStringToArray(strFieldValue)
+		#if string is not null, return array
+		if (strFieldValue != nil)
+			fieldArr =  strFieldValue.split(',').map(&:strip).reject(&:empty?).uniq 
+			return fieldArr                             
+		end     
 
-		    #if string is not null, return array
-		    if (strFieldValue != nil)
-		        fieldArr =  strFieldValue.split(',').map(&:strip).reject(&:empty?).uniq 
-		        return fieldArr                             
-		    end     
+		return [] #return empty array if string is nil
+		end
 
-		    return [] #return empty array if string is nil
-		 end
+		vrtArr = mapStringToArray(event.get('[event][payloadData][rolesCommaSeparated]'))
+		if vrtArr.length > 0                           
+			event.set('[event][payloadData][rolesCommaSeparated]', vrtArr)
+		end
 
-		 vrtArr = mapStringToArray(event.get('[event][payloadData][rolesCommaSeparated]'))
-		 if vrtArr.length > 0                           
-		    event.set('[event][payloadData][rolesCommaSeparated]', vrtArr) 
-		 end
+		event_array = []
+		event_type = event.get('[event][payloadData][eventType]')
+		idp_type = event.get('[event][payloadData][identityProviderType]')
+		auth_step_success = event.get('[event][payloadData][authStepSuccess]')
+        if (event_type == 'overall' || auth_step_success == false)
+			event_array.push('Overall')
+		end
+		if (event_type == 'step')
+			if (idp_type == 'Resident')
+				event_array.push('Resident')
+			end
+			if (idp_type == 'FEDERATED')
+				event_array.push('Federated')
+			end
+		end
+		event.set('[event][payloadData][event_type_filter]', event_array)
 	"
 	}
 
-	# debug timestamp
-	if ("" in [mtime]) {
-		date {
-    			match => [ "mtime" , "yyyy-MM-dd HH:mm:ss.SSS" ]
-    			target => "@timestamp"
-			locale => "en"
-			timezone => "Asia/Colombo"
+	# take log time as the timestamp
+	date {
+		match => [ "logtime" , "yyyy-MM-dd HH:mm:ss,SSS" ]
+		target => "@timestamp"
+	}
+
+	# session specific
+	if [eventType] == "session" {
+
+		# user agent filter to transform the userAgent string into seperated fields
+		useragent {
+			ecs_compatibility => disabled
+			source => "[event][payloadData][userAgent]"
+			target => "userAgentDetails"
 		}
+
+	} else if [eventType] == "auth" {
+
+		#geo ip filter to transform the IP address to location
+		geoip {
+			ecs_compatibility => disabled
+			source => "[event][payloadData][remoteIp]"
+		}
+
+		# geo IP failure to lookup or local address 
+		if "_geoip_lookup_failure" in [tags] {
+			mutate {
+				add_field => { "[geoip][country_name]" => "N/A" }
+			}
+		}
+		
 	}
+
 }
 
 output {
+
 	#seperating event type as auth and session via the eventtype
-	if [eventType] == "auth" {
+
+	if [eventType] == "auth"{
```

Comment:
Shall we add a space bethween "auth" and { 
`"auth" {`. Look for other places as well

Reply:
Fixed with https://github.com/wso2-extensions/identity-elk-integration/pull/14